lib/hello/accounts.ex

```diff
@@ -0,0 +1,86 @@
+defmodule Hello.Accounts do
+  @moduledoc """
+  The Accounts context.
+  """
+
+  import Ecto.Query, warn: false
+  alias Hello.Repo
+
+  alias Hello.Accounts.{User, Credential}
+
+
+  def list_users do
+    User
+    |> Repo.all()
+    |> Repo.preload(:credential)
+  end
+
+  def get_user!(id) do
+    User
+    |> Repo.get!(id)
+    |> Repo.preload(:credential)
+  end
+
+  def update_user(%User{} = user, attrs) do
+    user
+    |> User.changeset(attrs)
+    |> Ecto.Changeset.cast_assoc(:credential, with: &Credential.changeset/2)
+    |> Repo.update()
+  end
+
+  def create_user(attrs \\ %{}) do
+    %User{}
+    |> User.changeset(attrs)
```

Comment:
@JuanMaGiraldo what a changeset does?

Reply:
With Changesets, we can create and modify models. It's a structure that stores the changes which are then passed to `Ecto.Repo` to be persisted. In addition, Changeset provides us validators that will be verified before persisting.